test/hasManyThroughUpsert.test.js

```diff
@@ -0,0 +1,179 @@
+var request = require("supertest");
+var loopback = require('loopback')
+var expect = require('chai').expect;
+var JSONAPIComponent = require('../')
+var RSVP = require('rsvp')
+
+var app
+var Movie, Category, MovieCategoryAssoc;
+
+describe('hasManyThrough upsert', function () {
```

Comment:
Not sure if upsert is the right word here. Loopback has an upsert method eg. Post.upsert() but here we are just using post patch etc. upsert isn't used. Confused me initially. (Not a big deal)